user-management/user-details/script.js

```diff
@@ -292,5 +296,246 @@ function fetchNextTasks() {
   }
 }
 
+async function getUserSkills() {
+  try {
+    const res = await makeApiCall(
+      `${API_BASE_URL}/users/${userData?.id}/skills`,
+    );
+    if (res.status === 200) {
+      const data = await res.json();
+      userSkills = data.skills;
+      generateSkillsTabDetails(userSkills);
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+function generateSkillsTabDetails(skills) {
+  const div = createElement({
+    type: 'div',
+    classList: ['hidden-content', 'hide'],
+  });
+
+  if (skills.length) {
+    skills.forEach((skill) => {
+      const skillContainer = createElement({
+        type: 'div',
+        classList: ['skill-container', 'hidden-details'],
+      });
+      const skillName = createElement({ type: 'h3' });
+      const skillLevel = createElement({
+        type: 'span',
+      });
+      skillLevel.appendChild(createTextNode(`Level ${skill.levelValue}`));
+      skillName.appendChild(createTextNode(`${skill.tagName}`));
+      skillContainer.append(skillName, skillLevel);
+      div.append(skillContainer);
+    });
+  } else {
+    div.appendChild(createTextNode('No skills to show!'));
+  }
+
+  document.querySelector('.accordion-skills').append(div);
+}
+
+async function getUserAvailabilityStatus() {
+  try {
+    const res = await makeApiCall(
+      `${API_BASE_URL}/users/status/${userData?.id}`,
+    );
+    if (res.status === 200) {
+      const data = await res.json();
+      userStatusData = data.data;
+      generateAvalabilityTabDetails();
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+function generateAvalabilityTabDetails() {
+  if (userStatusData.currentStatus.state === 'OOO') {
+    generateUserNotAvailableDetails();
+  } else {
+    generateUserAvailableDetails();
```

Comment:
It is not mandatory that users will be available if their status is not OOO. It can be IDLE as well apart from active. I think we have missed this in design.